### PR TITLE
bug/MM-39875 : Fix warning printed after closing any ModalController modal

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
This PR fixes the console warning which shows up when a modal is closed. In this we are clearing out the modal key from the store. The files `reducers/views/modals.ts` and its test file are migrated to typescript as well.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39875

#### Related Pull Requests

~Has server changes (please link here)~
~Has mobile changes (please link here)~

#### Release Note

```release-note

```

### CODE REVIEW

**Review by AGMETEOR — APPROVED** [review]

File: reducers/views/modals.ts [hunk 1]
```diff
@@ -17,15 +20,13 @@ function modalState(state = {}, action) {
...
     case ActionTypes.MODAL_CLOSE:
-        return {
-            ...state,
-            [action.modalId]: {
-                open: false,
-                dialogProps: action.dialogProps,
-                dialogType: action.dialogType,
-            },
-        };
+        return Object.keys(state).reduce((previousModalState: ViewsState['modals']['modalState'], currentModalId) => {
```
Comment (hmhealey) [inline reducers/views/modals.ts:23]: Instead of looping through the state, we can use `Object.deleteProperty` to delete the key a bit easier

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: reducers/views/modals.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply reducers/views/modals.ts:23]: you mean [Reflect.deleteProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/deleteProperty) or [delete](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) , But in either case we are directly mutating the object itself which i feel is slightly not a better way when we are use pure non mutating way don't you think?

File: actions/views/modals.ts [hunk 2]
```diff
@@ -14,9 +14,15 @@ export function openModal<P>(modalData: ModalData<P>) {
...
 export function closeModal(modalId: string) {
     return {
         type: ActionTypes.MODAL_CLOSE,
         modalId,
     };
 }
+
+export type CloseModalReturnType = ReturnType<typeof closeModal>
+
+export type ActionsReturnType = OpenModalReturnType | CloseModalReturnType;
```
Comment (M-ZubairAhmed) [inline actions/views/modals.ts:28]: Replaced with `GenericAction` as we need to discuss this approach some more over at https://community.mattermost.com/core/pl/smb6hfdcuidu3rckja6y6p1gho

File: reducers/views/modals.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply reducers/views/modals.ts]: ✅  Updated as per request/discussion

**Review by hmhealey — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @M-ZubairAhmed running [the E2E tests here](https://automation-dashboard.vercel.app/cycles/24861107-b7dc-4c28-9216-df687f1807bb) I'll verify the report when ready.

**Comment by saturninoabril** [comment]
Branch updated to include recent fix on E2E that cause several tests to fail. Will run again on successful build - https://github.com/mattermost/mattermost-webapp/pull/9396

**Comment by jgilliam17** [comment]
Thanks @saturninoabril for re-running the E2Es. [New results here ](https://automation-dashboard.vercel.app/cycles/0e97c3a9-5e55-433a-ab8c-0478b3297704) - no related failures.

**Comment by jgilliam17** [comment]
Thanks @M-ZubairAhmed 
Tested, looks good to merge. 
- Verified opening and closing modals controlled by ModalController - no console errors.

**Pull request merged**